sdk/resourcemanager/Azure.ResourceManager/src/ArmClientOptions.cs

```diff
@@ -22,6 +22,11 @@ public sealed class ArmClientOptions : ClientOptions
         /// </summary>
         public ArmEnvironment? Environment { get; set; }
 
+        /// <summary>
+        /// Gets or sets API version profile. See https://github.com/Azure/azure-rest-api-specs/tree/main/profile#file-structure for data structure.
+        /// </summary>
+        public BinaryData ApiVersionProfile { get; set; }
```

Comment:
I wonder if this will create ambiguity with the `SetApiVersion` api since its not clear they do the same thing just in batch.

Perhaps we add an overload (ish) method like
`public void SetApiVersions(Uri profileUri)`
We might also consider a more explicit name like
`public void SetApiVersionsFromAzureStackProfile(Uri profileUri)`

I don't like the long winded name, but it makes it more clear we are expecting a very specific format here and you can't just point at any profile format you may have created yourself.

We might also consider a local file version of the API to support having the file already downloaded locally?

This method could read the profile and call `SetApiVersion` in a loop for each entry.